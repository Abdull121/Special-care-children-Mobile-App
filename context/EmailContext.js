import React, { createContext, useState, useContext } from "react";

const EmailContext = createContext(null);

export const EmailProvider = ({ children }) => {
    const [getEmail, setEmail] = useState("");

    return (
        <EmailContext.Provider value={{ getEmail, setEmail }}>
            {children}
        </EmailContext.Provider>
    );
};

export const useEmail = () => {
    const context = useContext(EmailContext);
    if (!context) {
        throw new Error("useEmail must be used within an EmailProvider");
    }
    return context;
};
