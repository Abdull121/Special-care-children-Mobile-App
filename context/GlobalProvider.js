import React, { createContext, useContext, useEffect, useState } from "react";

import authService from  "../Appwrite/auth";
import { Alert } from "react-native";

const GlobalContext = createContext();
export const useGlobalContext = () => useContext(GlobalContext);

const GlobalProvider = ({ children }) => {
  const [isLogged, setIsLogged] = useState(false);
  const [user, setUser] = useState([]);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    
    authService.getCurrentUser()
      .then((res) => {
        if (res) {
          setIsLogged(true);
          setUser(res);
        } else {
          setIsLogged(false);
          setUser([]);
        }
      })
      .catch((error) => {
        if (
          !error.message.includes("missing scope (account)") &&
          error.code !== 401 &&
          error.code !== 403
        ) {
          Alert.alert("Error", error.message);
        }
      })
      .finally(() => {
        setLoading(false);
      });
  }, []);

  return (
    <GlobalContext.Provider
      value={{
        isLogged,
        setIsLogged,
        user,
        setUser,
        loading,
      }}
    >
      {children}
    </GlobalContext.Provider>
  );
};

export default GlobalProvider;