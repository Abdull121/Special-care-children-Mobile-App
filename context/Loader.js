// context/LoadingContext.js cretaed for loading screen for google Auth
import React, { createContext, useContext, useState } from 'react';
import { ActivityIndicator, View, StyleSheet, Modal, Text, Dimensions, Image } from 'react-native';

// Get the device screen dimensions
const { width: SCREEN_WIDTH, height: SCREEN_HEIGHT } = Dimensions.get('window');

const LoadingContext = createContext();

export const LoadingProvider = ({ children }) => {
  const [loading, setLoading] = useState(false);
  const [loadingMessage, setLoadingMessage] = useState('Loading...');
  
  const showLoading = (message = 'Loading...') => {
    setLoadingMessage(message);
    setLoading(true);
  };
  
  const hideLoading = () => {
    setLoading(false);
  };
  
  return (
    <LoadingContext.Provider value={{ loading, showLoading, hideLoading }}>
      {children}
      <Modal visible={loading} transparent={false} animationType="fade">
        <View style={styles.fullScreenContainer}>
          {/* Your app logo */}
          <Image 
            source={require('../assets/images/children.png')}  // Update path to your logo
            style={styles.logo}
            resizeMode="contain"
          />
          
          <Text style={styles.appTitle}>SPECIAL CARE</Text>
          
          <View style={styles.loaderContainer}>
            <ActivityIndicator size="large" color="#0166FC" />
            <Text style={styles.loadingText}>{loadingMessage}</Text>
          </View>
        </View>
      </Modal>
    </LoadingContext.Provider>
  );
};

export const useLoading = () => useContext(LoadingContext);

const styles = StyleSheet.create({
  fullScreenContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#FFFFFF', // Use your app's background color
    padding: 20,
  },
  logo: {
    width: 120,
    height: 120,
    marginBottom: 20,
  },
  appTitle: {
    fontSize: 28,
    fontWeight: 'bold',
    color: '#0166FC', // Use your primary color
    marginBottom: 40,
    fontFamily: 'Poppins-Bold',
  },
  loaderContainer: {
    alignItems: 'center',
    justifyContent: 'center',
    marginTop: 20,
  },
  loadingText: {
    marginTop: 15,
    fontSize: 16,
    color: '#333',
    fontFamily: 'Poppins-Regular',
    textAlign: 'center',
  }
});