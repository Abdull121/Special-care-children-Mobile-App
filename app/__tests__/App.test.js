import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Index from '../../app/index';
import { useGlobalContext } from '../../context/GlobalProvider';
import { router } from 'expo-router';
import { within } from '@testing-library/react-native';


jest.mock('../../context/GlobalProvider', () => ({
  useGlobalContext: jest.fn(),
}));


jest.mock('expo-router', () => ({
  router: {
    push: jest.fn(),
  },
  Redirect: jest.fn(() => null),
}));

describe('Index Screen', () => {
  test('shows loading indicator when loading is true', () => {
    useGlobalContext.mockReturnValue({ loading: true, isLogged: false });

    const { getByTestId } = render(<Index />);
    expect(getByTestId('loading-indicator')).toBeTruthy();
  });

  test('redirects when user is logged in', () => {
    useGlobalContext.mockReturnValue({ loading: false, isLogged: true });

    const { UNSAFE_root } = render(<Index />);
    expect(within(UNSAFE_root).queryAllByText('*')).toHaveLength(0);
  });

  test('renders correctly when user is not logged in', () => {
    useGlobalContext.mockReturnValue({ loading: false, isLogged: false });

    const { getByText, getByTestId } = render(<Index />);
    
    expect(getByText('SPECIAL CARE')).toBeTruthy();
    expect(getByText(/Supporting children with special needs/i)).toBeTruthy();
    expect(getByTestId('get-started-button')).toBeTruthy();
  });

  test('navigates to login on button press', () => {
    useGlobalContext.mockReturnValue({ loading: false, isLogged: false });

    const { getByTestId } = render(<Index />);
    fireEvent.press(getByTestId('get-started-button'));

    expect(router.push).toHaveBeenCalledWith('/login');
  });

  test('matches snapshot', () => {
    useGlobalContext.mockReturnValue({ loading: false, isLogged: false });

    const tree = render(<Index />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
