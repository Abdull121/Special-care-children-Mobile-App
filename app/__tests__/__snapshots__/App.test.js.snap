// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Screen matches snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
>
  <Image
    className="flex-1 justify-center items-center"
    resizeMode="cover"
    source={
      {
        "testUri": "../../../assets/images/background.png",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <RNCSafeAreaView
    edges={
      {
        "bottom": "additive",
        "left": "additive",
        "right": "additive",
        "top": "additive",
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        {
          "height": "100%",
        }
      }
    >
      <View>
        <View
          className="w-full justify-center items-center px-6 h-[85vh]"
        >
          <Text
            className="font-heading text-secondary-Default text-[34px] mb-[35px]"
          >
            SPECIAL CARE
          </Text>
          <Image
            className="w-[231px] h-[214px]"
            resizeMode="contain"
            source={
              {
                "testUri": "../../../assets/images/children.png",
              }
            }
          />
          <Text
            className="text-center text-secondary-Default text-[15px] font-pregular leading-6 mt-[20px]"
          >
            Supporting children with special needs and their caregivers Comprehensive care management for physical, educational, and emotional needs.
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 4,
                "display": "flex",
                "flexDirection": "row",
                "height": 42,
                "justifyContent": "center",
                "marginTop": 41,
                "minHeight": 42,
                "opacity": 1,
                "width": 231,
              }
            }
            testID="get-started-button"
          >
            <Text
              className="text-center text-black-Default text-[14px] font-psemibold"
            >
              Get Started
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </RNCSafeAreaView>
</View>
`;
